util/parsenum.h

```diff
@@ -61,14 +61,16 @@ _Pragma("clang diagnostic pop")
 		PARSENUM_PROLOGUE					\
 		errno = 0,						\
 		(((*(x)) = 1, (*(x)) /= 2) > 0)	?			\
-			((*(x)) = parsenum_float((s), (min), (max))) :	\
+			((*(x)) = parsenum_float((s), (double)(min),	\
+			    (double)(max))) :				\
 		(((*(x)) = -1) <= 0) ?					\
 			((*(x)) = parsenum_signed((s),			\
 			    (*(x) <= 0) ? (min) : 0,			\
 			    (*(x) <= 0) ? (max) : 0)) :			\
 			(((*(x)) = parsenum_unsigned((s),		\
 			    (min) <= 0 ? 0 : (min),			\
-			    (max) >= *(x) ? *(x) : (max))),		\
+			    (double)(max) >= (double)(*(x)) ?		\
```

Comment:
See the comment in 13c93b1:

> Since (max) could be INFINITY, we cannot cast this to an int: the C99 standard
says that's undefined behaviour [in 6.3.1.4 Real floating and integer].